tests/pytype_test.py

```diff
@@ -131,14 +135,43 @@ def find_stubs_in_paths(paths: Sequence[str]) -> list[str]:
     return filenames
 
 
+def get_missing_modules(files_to_test: Sequence[str]) -> Iterable[str]:
+    """Gets module names provided by typeshed-external dependencies.
+
+    Some typeshed stubs depend on dependencies outside of typeshed. Since pytype
+    isn't able to read such dependencies, we instead declare them as "missing"
+    modules, so that no errors are reported for them.
+    """
+    stub_distributions = set()
+    for fi in files_to_test:
+        parts = fi.split(os.sep)
+        try:
+            idx = parts.index("stubs")
+        except ValueError:
+            continue
+        stub_distributions.add(parts[idx + 1])
+    missing_modules = set()
+    for distribution in stub_distributions:
+        for pkg in utils.read_dependencies(distribution).external_pkgs:
```

Comment:
I think you need to do a little bit of parsing here, since the strings returned by `read_dependencies(distribution).external_pkgs` might include version constraints (e.g. https://github.com/python/typeshed/pull/9459/files#diff-a8f442db0372b4c61c228311fea276c35a7350c083cfcc08a32d2b22164ecee7R3).

(You'll need to add `from packaging.requirements import Retirement` to the top of the file.)

After this PR is merged, I'll try to file a PR to clarify exactly what's returned by `read_dependencies(distribution)`, since this does seem like a bit of a "gotcha" at the moment :)

```suggestion
        for req in utils.read_dependencies(distribution).external_pkgs:
            pkg = Requirement(req).name
```